local/include/core/http.h

```diff
@@ -241,6 +252,9 @@ class HttpHeader
    */
   void merge(HttpFields const &other);
 
+  /// Whether this is an HTTP/3 message.
```

Comment:
This is probably the point at which to switch from `bool` to an enumeration.

Reply:
Yeah, good point.

While I'm at it, I'll also reduce the will_receive_request and will_receive_response booleans to one with state variable with two functions of those names.